pkg/resources/kafka/kafka.go

```diff
@@ -102,9 +113,42 @@ func New(client client.Client, directClient client.Reader, cluster *v1beta1.Kafk
 			DirectClient: directClient,
 			KafkaCluster: cluster,
 		},
-		kafkaClientProvider: kafkaClientProvider,
+		kafkaClientProvider:        kafkaClientProvider,
+		CruiseControlScalerFactory: scale.ScaleFactoryFn(),
 	}
 }
+
+func getBrokerAzMap(cluster *v1beta1.KafkaCluster) map[int32]string {
+	brokerAzMap := make(map[int32]string)
+	for _, broker := range cluster.Spec.Brokers {
+		readOnlyConfigs, err := properties.NewFromString(broker.ReadOnlyConfig)
+		if err == nil {
+			brokerRack, brokerRackConfigFound := readOnlyConfigs.Get("broker.rack")
+			if brokerRackConfigFound {
+				brokerAzMap[broker.Id] = brokerRack.Value()
+			}
+		}
+	}
+	// if incomplete broker AZ information, consider all brokers as being in different AZs
+	if len(brokerAzMap) != len(cluster.Spec.Brokers) {
+		for _, broker := range cluster.Spec.Brokers {
+			brokerAzMap[broker.Id] = fmt.Sprintf("%d", broker.Id)
+		}
+	}
+	return brokerAzMap
+}
+
+func getBrokerRack(readOnlyConfig string) string {
```

Comment:
Question: It looks like the latest implementation doesn't require this func, can we remove it since it's not being used?

Reply:
Oh, good catch! Forgot about it. Will address in another commit, thanks for paying attention!